### PR TITLE
feat: fix archived reports

### PR DESCRIPTION
**Pull request by kphan32**
## PR Title/Tagline

Issue Number(s): #200 

- Fix archived report PDF not showing
- Fix archived reports not persisting image selections
- Fix archived reports not persisting notes
- Fix ordering of notes on report PDFs

### Checklist

- [ ] Changes to standard cards in the report builder actually propagate to the final report.
- [ ] If we view the edited cards in the "Completed Reports" we see that the change still exists.
- [ ] Editing a standard from the report builder should not affect the original standard card from digital library

### How to Test

- Run through creating a report, archiving it, and editing it again

### CODE REVIEW

File: server/mongodb/models/User.js [hunk 1]
```diff
@@ -21,7 +20,41 @@ const UserSchema = new Schema({
...
+                    type: Date,
+                    required: true,
+                    default: Date.now(),
+                  },
+                },
+              ],
+              default: [],
+            },
+          },
+        ],
```
Comment (kphan32) [inline server/mongodb/models/User.js:57]: whatever idiot implemented report-specific notes (me lol) didn't do the same for archived reports. also, archived reports never saved image selections either! this PR brings archived reports in-line with active reports.

File: src/components/ArchivedReportCard/StandardCard.jsx [hunk 2]
```diff
@@ -19,15 +19,21 @@ const StandardCard = ({ title, images, criteria }) => {
...
-              height={imageSize}
-              width={imageSize}
-              key={index}
-            />
-          ))}
+          {images
+            .filter((_, i) => {
+              if (!imgSelections) return true;
+              return imgSelections[i];
+            })
```
Comment (kphan32) [inline src/components/ArchivedReportCard/StandardCard.jsx:26]: use the persisted image selections saved on the archived report

File: src/components/Notes/Note.jsx [hunk 3]
```diff
@@ -45,8 +45,8 @@ const Note = ({
       setCurrNote(note.body);
       return;
     }
-    const newNotes = notes.map((n, idx) => {
-      if (idx === currNoteIdx) {
+    const newNotes = notes.map((n) => {
+      if (n._id === note._id) {
```
Comment (kphan32) [inline src/components/Notes/Note.jsx:49]: theres a lot of array reversing happening in the front-end, so i simplified this to just check for IDs instead

File: src/components/ReportDocumentPDF/ReportDocumentPDF.jsx [hunk 4]
```diff
@@ -146,20 +146,23 @@ const ReportDocumentPDF = ({ selectedReport, sels }) => {
...
-                              day: "numeric",
-                              year: "numeric",
-                            })}
-                          </Text>
-                        </View>
-                      );
-                    })}
+                    {notes
+                      .slice()
+                      .reverse()
```
Comment (kphan32) [inline src/components/ReportDocumentPDF/ReportDocumentPDF.jsx:151]: real change are these two lines to flip the order of the notes (this is how notes are rendered in the report builder)

**Review by avayedawadi — APPROVED** [review]

**Pull request merged**